tests/test_h2_client.c

```diff
@@ -639,7 +639,9 @@ TEST_CASE(h2_client_stream_err_receive_data_before_headers) {
 
     /* validate that stream sent RST_STREAM */
     ASSERT_SUCCESS(h2_fake_peer_decode_messages_from_testing_channel(&s_tester.peer));
-    struct h2_decoded_frame *rst_stream_frame = h2_decode_tester_latest_frame(&s_tester.peer.decode);
+    /* last two frames will be window updates */
+    struct h2_decoded_frame *rst_stream_frame =
+        h2_decode_tester_get_frame(&s_tester.peer.decode, h2_decode_tester_frame_count(&s_tester.peer.decode) - 3);
     ASSERT_INT_EQUALS(AWS_H2_FRAME_T_RST_STREAM, rst_stream_frame->type);
```

Comment:
Whoa, couple of things here:
First, it's a crime to send anything to a stream after RST_STREAM
> An endpoint that receives any frame other than PRIORITY after receiving a RST_STREAM MUST treat that as a stream error (Section 5.4.2) of type STREAM_CLOSED

Second, we should try if possible not to make the tests suuper dependent on the order of frames. Since as we see here, the connection can now send WINDOW_UPDATE frames that shouldn't affect stream frames.

There are some helper functions like `h2_decode_tester_find_stream_frame()` you should use to find the RST_STREAM, instead of looking at an absolute index. There's another find() function that finds a frame of the specified type, but doesn't care about stream_id. We could add another that finds frames of any type, but with the specified stream_id, something like `h2_decode_tester_find_stream_frame_any_type()`